packages/block-editor/src/store/test/private-selectors.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * Internal dependencies
+ */
+import { isBlockInterfaceHidden } from '../private-selectors';
+
+describe( 'private selectors', () => {
+	describe( 'isBlockInterfaceHidden', () => {
+		it( 'should return the true if toggled true in state', () => {
+			const state = {
+				isBlockInterfaceHidden: true,
+			};
+
+			expect( isBlockInterfaceHidden( state ) ).toBe( true );
```

Comment:
I like to test actions and selectors by creating an actual registry and confirming the entire system is well-behaved, e.g. 

```js

	const registry = createRegistry();
	registry.register( blockEditorStore );
	
	await unlock(registry.dispatch( blockEditorStore ))
		. hideBlockInterface();

	// Check that blocks store contains the new block.
	const isHidden = 
		unlock(registry.select( blockEditorStore ))
			. isBlockInterfaceHidden();
	expect( isHidden ).toBeTruthy();
```

This reflects how the code is actually used and signals a useful error in case anything is wrong in the flow.

On the other hand, testing a selector with a mock state will only throw an error if the state property name is updated – even if the action, reducer, and the selector itself have been updated properly.

Reply:
I like that approach too. It's a bit beyond the scope of this PR though, so I'll look into changing this separately.